ticketbuyer/mempool.go

```diff
@@ -69,16 +69,22 @@ func (t *TicketPurchaser) ownTicketsInMempool() (int, error) {
 	// next block attached yet.
 	var curStakeInfo *wallet.StakeInfoData
 	var err error
+out:
 	for i := 0; i < stakeInfoReqTries; i++ {
-		curStakeInfo, err = t.wallet.StakeInfo()
-		if err != nil {
-			log.Tracef("Failed to fetch stake information "+
-				"on attempt %v: %v", i, err.Error())
-			time.Sleep(stakeInfoReqTryDelay)
-			continue
-		}
-		if err == nil {
-			break
+		select {
+		case <-t.quit:
+			return 0, nil
+		default:
+			curStakeInfo, err = t.wallet.StakeInfo()
```

Comment:
Not a big fan that this can block on the RPC (and even the sleep if it errors) when the quit channel was closed, until it is read next.  I would run it from a goroutine and use the select to block on their the async result or the quit channel.